File: cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/event/platform/GetVirtualMachineTypesResult.java
package com.sequenceiq.cloudbreak.cloud.event.platform;

import com.sequenceiq.cloudbreak.cloud.event.CloudPlatformRequest;
import com.sequenceiq.cloudbreak.cloud.event.CloudPlatformResult;
import com.sequenceiq.cloudbreak.cloud.model.PlatformVirtualMachines;

public class GetVirtualMachineTypesResult extends CloudPlatformResult {
    private PlatformVirtualMachines platformVirtualMachines;

    public GetVirtualMachineTypesResult(CloudPlatformRequest<?> request, PlatformVirtualMachines platformVirtualMachines) {
        super(request);
        this.platformVirtualMachines = platformVirtualMachines;
    }

    public GetVirtualMachineTypesResult(String statusReason, Exception errorDetails, CloudPlatformRequest<?> request) {
        super(statusReason, errorDetails, request);
    }

    public PlatformVirtualMachines getPlatformVirtualMachines() {
        return platformVirtualMachines;
    }
}

File: cloud-reactor-api/src/main/java/com/sequenceiq/cloudbreak/cloud/event/platform/GetVirtualMachineTypesResult.kt
package com.sequenceiq.cloudbreak.cloud.event.platform

import com.sequenceiq.cloudbreak.cloud.event.CloudPlatformRequest
import com.sequenceiq.cloudbreak.cloud.event.CloudPlatformResult
import com.sequenceiq.cloudbreak.cloud.model.PlatformVirtualMachines

class GetVirtualMachineTypesResult : CloudPlatformResult<CloudPlatformRequest<Any>> {
    val platformVirtualMachines: PlatformVirtualMachines

    constructor(request: CloudPlatformRequest<*>, platformVirtualMachines: PlatformVirtualMachines) : super(request) {
        this.platformVirtualMachines = platformVirtualMachines
    }

    constructor(statusReason: String, errorDetails: Exception, request: CloudPlatformRequest<*>) : super(statusReason, errorDetails, request) {
    }
}
